### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'folderId')

### ISSUE THREAD

**Issue by arifje** [spicywebau/craft-embedded-assets#240]
### Bug Description

Since our update to Craft 4.5.1 and Embedded Assets 3.1.6, we can't save the embeds anymore.

```
main.ts:67 Uncaught TypeError: Cannot read properties of undefined (reading 'folderId')
    at o.c [as _currentGetActionTarget] (main.ts:67:32)
    at o._getActionTarget (EmbeddedAssets.ts:76:17)
    at h._getActionTarget (EmbeddedAssets.ts:17:39)
    at h.save (Form.ts:137:72)
    at HTMLDivElement.<anonymous> (Modal.ts:57:45)
    at HTMLDivElement.dispatch (jquery.js?v=1692888808:2:43392)
    at v.handle (jquery.js?v=1692888808:2:41387)
```

### Steps to reproduce

- Create an entry
- Add an embed
- Copy paste the URL of a Youtube video
- The preview modal pops up
- Pressing save generates Javascript error
- Asset is not saved

### Embedded Assets version

3.1.6

### Craft CMS version

4.5.1

**Comment by joshuabaker** [comment]
Could be just that we need the extra check or optional chain as referencing properties of a non-object will throw.

Here’s a possible drop-in replacement for the `getActionTarget` method.

```ts
  const getActionTarget: () => Object = () => {
    if (Array.isArray(this.sourcePath)) {
      // Craft 4.4 subfolder compatibility
      const currentFolder = this.sourcePath[this.sourcePath.length - 1]

      if (currentFolder?.folderId) {
        return {
          targetType: 'folder',
          targetId: currentFolder.folderId
        }
      }
    }

    const [targetType, targetUid] = this.sourceKey.split(':')

    if (targetType && targetUid) {
      return { targetType, targetUid }
    }

    return {}
  }
```

I’ve also stripped the `typeof` checks in favour of clearer and type-checked conditions. It’s absolutely a personal preference, but I think the conditions and destructuring assignment syntax make it a bit more readable.

All of this is assuming that `currentFolder` should ever be nullish.

**Comment by ttempleton** [comment]
This is working as expected for me, e.g. the following screenshot shows me logging the `sourcePath` property:

<img width="579" alt="Screenshot 2023-08-30 at 20 25 54" src="https://github.com/spicywebau/craft-embedded-assets/assets/19421878/4bb68d04-befb-4c14-aac5-d95dbadea10b">

Not really willing to make changes without being able to reproduce the error, since I don't want to make a change that breaks something else, and `sourcePath` is actually a `Craft.AssetIndex` property and I'm unsure what case causes it to be empty. Any extra information that could help identify the cause (e.g. a specific field configuration if this only happens when editing entries) is appreciated.

**Comment by joshuabaker** [comment]
Thanks, @ttempleton. Where are you seeing that array you’re referencing?

**Comment by joshuabaker** [comment]
<details>
<summary>Field settings</summary>
<img src="https://github.com/spicywebau/craft-embedded-assets/assets/160484/13e66954-01da-44b0-bad5-b19792ec981d" alt="field settings" />
</details>

**Comment by arifje** [comment]
Our field config;

![Screenshot 2023-08-30 at 13 16 07](https://github.com/spicywebau/craft-embedded-assets/assets/1286410/51a40dc3-7055-44b3-bb0c-3424932c520b)

**Comment by ttempleton** [comment]
Thanks both - I can reproduce the error and it is related to the use of variables when restricting assets to a single location.

Just want to confirm that this has only started happening when upgrading to Craft 4.5? This is just for the purpose of updating the changelog.

**Comment by joshuabaker** [comment]
Thanks, @ttempleton. Glad we were able to help diagnose this.

I was seeing this with earlier versions (e.g. Craft `4.4.17`).

**Comment by ttempleton** [comment]
Thanks - just released 3.1.7 with the fix.

**Comment by joshuabaker** [comment]
Amazing! Thanks, @ttempleton. Can confirm this is up and running again for us. 🙏🏻
